kornia/augmentation/container/augment.py

```diff
@@ -332,13 +341,23 @@ def _validate_args_datakeys(self, *args: DataType, data_keys: List[DataKey]) ->
     def _arguments_preproc(self, *args: DataType, data_keys: List[DataKey]) -> List[DataType]:
         inp: List[DataType] = []
         for arg, dcate in zip(args, data_keys):
-            if DataKey.get(dcate) in _IMG_MSK_OPTIONS:
+            if DataKey.get(dcate) in _IMG_OPTIONS:
+                arg = cast(Tensor, arg)
+                self.input_dtype = arg.dtype
```

Comment:
I think we should use another for-loop to find the `input_dtype` first. Otherwise, it may happen if the `INPUT` type comes after `MASK`.

Reply:
Doesn't this if block make sure that does not happen?  It ensures INPUT DataKey must be present and will always be the first DataKey to be evaluated.
 https://github.com/kornia/kornia/blob/e461f92ff9ee035d2de2513859bee4069356bc25/kornia/augmentation/container/augment.py#L240